src/dune_rules/findlib/ocamlfind.ml

```diff
@@ -0,0 +1,92 @@
+open Import
+
+type t =
+  { config : Findlib.Config.t
+  ; ocamlpath : Path.t list
+  ; which : string -> Path.t option Memo.t
+  ; toolchain : string option
+  }
+
+let ocamlpath_sep =
+  if Sys.cygwin then (* because that's what ocamlfind expects *)
+    ';'
+  else Bin.path_sep
+
+let path_var = Bin.parse_path ~sep:ocamlpath_sep
+
+let ocamlpath env =
+  match Env.get env "OCAMLPATH" with
+  | None -> []
+  | Some s -> path_var s
+
+let set_toolchain t ~toolchain =
+  match t.toolchain with
+  | None ->
+    { t with
+      config = Findlib.Config.toolchain t.config ~toolchain
+    ; toolchain = Some toolchain
+    }
+  | Some old_toolchain ->
+    Code_error.raise "Ocamlfind.set_toolchain: cannot set toolchain twice"
+      [ ("old_toolchain", Dyn.string old_toolchain)
+      ; ("toolchain", Dyn.string toolchain)
+      ]
+
+let conf_path t =
+  match Findlib.Config.get t.config "path" with
```

Comment:
is this "path" config something that we've added to dune, or is it from the findlib standard?

I don't remember anymore.

Reply:
this is `path = " ... "` or `path(TOOLCHAIN) = "..."` in findlib.conf. It's the ocaml library path (which `OCAMLPATH` appends to).